src/signer.ts

```diff
@@ -277,8 +277,14 @@ export class URLSigner {
     const credentialScope = `${datestamp}/auto/storage/goog4_request`;
 
     const sign = async () => {
-      const credentials = await this.authClient.getCredentials();
-      const credential = `${credentials.client_email}/${credentialScope}`;
+      let client_email;
+      if ((this.authClient as any).jsonContent?.client_email == null) { // eslint-disable-line
```

Comment:
I don’t believe we should access those internal properties from this library. “getCredentials()” should be returning the cached credentials, rather than re-fetching them each time. If it’s not, we should look into adding that behavior over there.

Reply:
Yes, you are right. That is a better fix. Looking into this, is actually appears that this _was_ fixed in google auth library upstream. So all I needed to do was update the library